src/utils/workflow_utils/src/workflow_utils.c

```diff
@@ -1804,57 +1804,86 @@ bool workflow_set_sandbox(ADUC_WorkflowHandle handle, const char* sandbox)
     return true;
 }
 
-// Workfolder =  [root.sandboxfolder]  "/"  ( [parent.workfolder | parent.id]  "/" )+  [handle.workfolder | handle.id]
-char* workflow_get_workfolder(const ADUC_WorkflowHandle handle)
+// Returns the download base directory
+char* workflow_get_workflow_base_dir(const ADUC_WorkflowHandle handle)
 {
     char dir[1024] = { 0 };
-
-    char* pwf = NULL;
     char* ret = NULL;
-    char* id = workflow_get_id(handle);
-
-    // If workfolder explicitly specified, use it.
-    char* wf = workflow_get_string_property(handle, WORKFLOW_PROPERTY_FIELD_WORKFOLDER);
-    if (wf != NULL)
-    {
-        Log_Debug("Property '%s' not NULL - returning cached workfolder '%s'", WORKFLOW_PROPERTY_FIELD_WORKFOLDER, wf);
-        free(id);
-        return wf;
-    }
+    char* pwf = NULL;
 
-    // Return ([parent's workfolder] or [default sandbox folder]) + "/" + [workflow id];
     ADUC_WorkflowHandle p = workflow_get_parent(handle);
```

Comment:
what does workflow_get_parent do?  parent of what?

Reply:
If there are nested workflows this allows the parent directory to be used for the processing of child workflows.